### PR TITLE
3525: Properly hide list; update list in any case (empty)

### LINKED ISSUE #3525: Delete last filter: list is not empty
Unter Account settings > Filters
If you delete the last filter that filter is still shown in the list.

* * * *
- Tusky Version: Current develop (16e1a)

### PR DESCRIPTION
**Pull request by Lakoja**
Fixes #3525

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt [hunk 1]
```diff
@@ -81,9 +84,12 @@ class FiltersActivity : BaseActivity(), FiltersListener {
                             binding.messageView.show()
                         } else {
                             binding.messageView.hide()
-                            binding.filtersList.adapter = FiltersAdapter(this@FiltersActivity, state.filters)
```
Comment (Lakoja) [inline app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt:84]: Must be done in any case: Also an empty list is valid.

File: app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt [hunk 2]
```diff
@@ -81,9 +84,12 @@ class FiltersActivity : BaseActivity(), FiltersListener {
...
                             binding.filtersList.show()
                         }
+
+                        if (binding.filtersList.adapter == null || (binding.filtersList.adapter as FiltersAdapter).filters != state.filters) {
```
Comment (Tak) [inline app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt:90]: Why not just `?.`?

**Review by Tak — APPROVED** [review]

File: app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt (hunk 2 above)
Comment (Lakoja) [reply app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt:90]: How exactly would that look?

Comment (Tak) [reply app/src/main/java/com/keylesspalace/tusky/components/filters/FiltersActivity.kt:90]: Sorry, missed this reply in my notifications.
I'm thinking like `(binding.filtersList.adapter as FiltersAdapter)?.filters != state.filters` - but it's just a minor nitpick

**Pull request closed without merging**